### ISSUE TITLE
Replacing database file created by expo-sqlite generate Error: attempt to write a readonly database

### ISSUE THREAD

**Issue by allandiego** [expo/expo#8109]
### Environment
Expo CLI 3.20.1 environment info:
    System:
      OS: Windows 10 10.0.18362
    Binaries:
      Node: 13.13.0 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.4 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 6.9.0 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      expo: ^37.0.0 => 37.0.8
      react: 16.9.0 => 16.9.0
      react-dom: 16.9.0 => 16.9.0
      react-native: https://github.com/expo/react-native/archive/sdk-37.0.1.tar.gz => 0.61.4
      react-native-web: ^0.11.7 => 0.11.7

Android/  Standalone

### Steps to Reproduce
Long version:
https://forums.expo.io/t/expo-sqlite-backup-restore-error-readonly-database-code-1032/36958

TL;DR:
- open a database with expo SQLite
- create table / insert data
- close connection
- replace the database file created automatically  in ${FileSystem.documentDirectory}SQLite/ for another database with the same name but direfent data like 1 more row inserted
- try to open the new database file and do a transction insert/delete
- Error: attempt to write a readonly database (code 1032) 

### Expected Behavior
Expected to be able to replace the file with a backup one and sqlite transactions continue to works

If you wont let the sqlite create the database file automatically and place backup in ${FileSystem.documentDirectory}SQLite/ before any openDatabase() and then open it, backup/restore process works, i can close connection and replace the file freely without errors and insert delete works.

I noticed that placing file before and then opening connection dont create the -journal file, maybe the issue can be related with that file
https://www.sqlite.org/tempfiles.html

### Reproducible Demo
https://snack.expo.io/@adcr/sqlite-file-replace-error

**Comment by byCedric** [comment]
Hi @sstur! Thanks for the report 😄 I think I found your issue. The main issue that causes the "attempt to write a readonly database" is the non-closed database references. You open the database with `const db = ...`, but never close it. This causes the process to keep it open and may cause issues like this.

Now to fix that, we need to solve another issue. If you run `db.transaction` and `db._db.close` right after each other, the closing isn't waiting for the transaction to finish. I wrapped them into a Promise to use `await` on it.

I [created a new snack based on yours](https://snack.expo.io/@bycedric/sqlite-file-replace-error), it works locally but seems to have some issues in Snack. Could you try running it locally?

Hope it helps!

**Comment by allandiego** [comment]
Hello @byCedric, thanks for your answer, with your explanation i was able to figure out my original problem with the expo driver for typeorm lib, apparently the function db._db.close(); was implemented not too long ago and it was not begin called in typerom getConnection().close().

I think it would be interesting to improve the expo-sqlite docs for example this db._db.close(); it's kind of hidden function

**Comment by byCedric** [comment]
You are right, we probably want to expose the `close` method too. It's part of the API you should be able to control.

**Comment by Krishna205101** [comment]
I tried it the same way. But instead of opening  the downloaded database it is creatring a new database with a temp extension in name and performing operations in it.
